### PR TITLE
special tag trigger release

### PR DESCRIPTION
**Pull request by qingwusunny**

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -29,9 +30,36 @@ jobs:
...
+        run: if [[ ${github.ref} =~ ^refs\/tags\/[0-9]+\.[0-9]+\.[0-9]+$ ]]; then echo "NEED_RELEASE=true" >> $GITHUB_ENV; else echo "NEED_RELEASE=false" >> $GITHUB_ENV; fi
+
+    outputs:
+      ReleaseTag: ${{ env.IMAGE_TAG }}
+      NeedRelease: ${{ env.NEED_RELEASE }}
+
+  publish-release:
+    needs: [build]
+    runs-on: ubuntu-latest
+    if: ${{ needs.build.outputs.NeedRelease == 'true' }}
```
Comment (wu-cl) [inline .github/workflows/build.yaml:50]: jobs are parallel

Comment (qingwusunny) [reply .github/workflows/build.yaml:60]: use needs

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -35,3 +35,41 @@ jobs:
...
+            echo "NEED_RELEASE=true" >> $GITHUB_ENV
+          else 
+            echo "NEED_RELEASE=false" >> $GITHUB_ENV
+          fi
+          
+          exists=$(curl -s -H "Accept: text/html" https://github.com/$GITHUB_REPOSITORY/releases/tag/${{ env.IMAGE_TAG }} -w %{http_code} -o /dev/null)
+          if $exists == 200; then 
+            echo "release ${{ env.IMAGE_TAG }} has been exists"
+            echo "NEED_RELEASE=false" >> $GITHUB_ENV;
+          fi
```
[the remark this answers is not in the document]
Comment (qingwusunny) [reply .github/workflows/build.yaml:51]: do you mean check $exists == 404 ?

[the remark this answers is not in the document]
Comment (qingwusunny) [reply .github/workflows/build.yaml:51]: This check is not reliable, delete it.

**Review by KeyongSun — APPROVED** [review]

**Pull request merged**